block-storage/src/main/scala/coop/rchain/blockstorage/LMDBBlockStore.scala

```diff
@@ -84,6 +84,25 @@ class LMDBBlockStore[F[_]] private (val env: Env[ByteBuffer], path: Path, blocks
             }
     } yield ret
 
+  override def find(p: BlockHash => Boolean): F[Seq[(BlockHash, BlockMessage)]] =
+    for {
+      _ <- metricsF.incrementCounter(MetricNamePrefix + "find")
+      ret <- withReadTxn { txn =>
+              blocks
+                .iterate(txn)
+                .iterable()
```

Comment:
Why turn into an `Iterable` here? I'm not too sure about the LMDB API overall, but I would wait until after the filter is applied.